src/powershell/Public/Remove-FinOpsCostExport.ps1

```
@@ -0,0 +1,169 @@
+#>
+
+function Remove-FinOpsCostExport
+{
+  [CmdletBinding(SupportsShouldProcess)]
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Name,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Scope,
+
+    [Parameter()]
+    [switch]
+    $RemoveData,
+
+    [Parameter()]
+    [string]
+    $APIVersion = '2023-04-01-preview'
+  )
+
+  try
+  {
+    $additionalParamaters = @{
+      scope = $scope
+    }
+    $payload = ConvertTo-Json -InputObject $additionalParamaters -Depth 100
+
+    # TODO : Validate parameters and resources before proceeding
+    $httpResponse = Invoke-AzRestMethod `
+      -ResourceProviderName "Microsoft.CostManagement" `
+      -ResourceType "exports" `
+      -Name $Name `
+      -ApiVersion $ApiVersion `
+      -Method "GET" `
+      -Payload $payload
+
+    if ($httpResponse.StatusCode -eq 200)
+    {
+      # Export deleted successfully
+    }
+    elseif ($httpResponse.StatusCode -eq 404)
+    {
+      # Not found
+    }
+    else
+    {
+      # Error response describing why the operation failed.
+      throw "Not Exists: Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+    }
+
+    if ($PSCmdlet.ShouldProcess($Name, 'DeleteCostExport'))
+    {
+
+      # Using the REST API to delete the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "DELETE" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+        # Export deleted successfully
+      }
+      elseif ($httpResponse.StatusCode -eq 404)
+      {
+        # Not found - nothing to delete
+      }
+      else
+      {
+        # Error response describing why the operation failed.
+        throw "Delete Cost Management Export operation failed with message: '$($httpResponse.Content)'"
```

Is there a standard practice for returning the error code and error message in an object? I know that's important for REST calls. I'm assuming it applies to PowerShell as well, but not sure. @aromano2?

---

Yea for rest calls we should just bubble the exception up. Good catch

---

Not sure what is needed here.